src/stc/PlatWX.cpp

```diff
@@ -2541,20 +2521,31 @@ void ListBoxImpl::RegisterImageHelper(int type, wxBitmap& bmp)
 }
 
 void ListBoxImpl::RegisterImage(int type, const char *xpm_data) {
-    wxMemoryInputStream stream(xpm_data, strlen(xpm_data)+1);
-    wxImage img(stream, wxBITMAP_TYPE_XPM);
+    wxXPMDecoder dec;
+    wxImage img;
+
+    // This check is borrowed from src/stc/scintilla/src/XPM.cpp.
+    // Test done is two parts to avoid possibility of overstepping the memory
+    // if memcmp implemented strangely. Must be 4 bytes at least at destination.
+    if ( (0 == memcmp(xpm_data, "/* X", 4)) &&
+         (0 == memcmp(xpm_data, "/* XPM */", 9)) )
+    {
+        wxMemoryInputStream stream(xpm_data, strlen(xpm_data)+1);
+        img = dec.ReadFile(stream);
+    }
+    else
+        img = dec.ReadData(reinterpret_cast<const char* const*>(xpm_data));
+
     wxBitmap bmp(img);
-    RegisterImageHelper(type, bmp);
+    RegisterImageHelper(type,bmp);
```

Comment:
Extremely minor, but there doesn't seem any reason to remove this space...
```suggestion
    RegisterImageHelper(type, bmp);
```